seastar/src/sleep.cc

```diff
@@ -3,5 +3,21 @@
 namespace seastar_ffi {
 namespace sleep {
 
+VoidFuture sleep_secs(int32_t duration) {
+    co_await seastar::sleep(std::chrono::seconds(duration));
+}
+
+VoidFuture sleep_millis(int32_t duration) {
+    co_await seastar::sleep(std::chrono::milliseconds(duration));
+}
+
+VoidFuture sleep_micros(int32_t duration) {
+    co_await seastar::sleep(std::chrono::microseconds(duration));
+}
+
```

Comment:
I don't think that we need those. A single `sleep_nanos` should be sufficient. 2^64 nanoseconds is about 500-600 years, so `sleep_nanos` should be sufficient (for some time, at least).

Reply:
Later, we can use `sleep_nanos` to build nicer, idiomatic abstractions - for example, introduce a `SteadyDuration` type which can be created from a number of seconds/milliseconds/etc. It would keep the duration as a number of nanoseconds so `sleep_nanos` could suffice.

Reply:
After introducing duration types this issue does not exist anymore.

Reply:
> Later, we can use `sleep_nanos` to build nicer, idiomatic abstractions - for example, introduce a `SteadyDuration` type which can be created from a number of seconds/milliseconds/etc. It would keep the duration as a number of nanoseconds so `sleep_nanos` could suffice.

I defined instant and duration types for all 3 clocks specializations.

There was a suggestion in a review of #32 that we could use `std::time::Duration`. It could be really easy to implement but I decided that creating our own duration types is worth the effort. I'll try to explain why.

Panicking when too large `std::time::Duration` is given wouldn't be a big problem but I think that something like *don't use durations overflowing `i64`* appearing in documentation would just look really bad. It would be pretty obvious that using `std::time::Duration` was just a lazy alternative to creating a new type.

Moreover, duration types appear in many places (sleep, timer, semaphore) and might appear somewhere else when this project grows. Creating our own types gives us more control and if (for whatever reason) a change related to durations is needed, then we just have to change the duration types instead of all parts of code that use durations.

> I don't think that we need those. A single `sleep_nanos` should be sufficient. 2^64 nanoseconds is about 500-600 years, so `sleep_nanos` should be sufficient (for some time, at least).

After introducing duration types this issue does not exist anymore.